<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1090</width>
    <height>685</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_5">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,8">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1,1,1,1,1,1,1">
          <item>
           <spacer name="verticalSpacer_5">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayout_4" rowstretch="1,0">
            <item row="1" column="0">
             <widget class="QPushButton" name="refreshDevListButton">
              <property name="text">
               <string>刷新列表</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="2">
             <widget class="QComboBox" name="deviceListCombo"/>
            </item>
            <item row="1" column="1">
             <widget class="QPushButton" name="openDevButton">
              <property name="text">
               <string>打开</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayout_6" columnstretch="3,5,1">
            <item row="1" column="0" colspan="3">
             <widget class="QPushButton" name="refreshSpectrumButton">
              <property name="text">
               <string>单次获取</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QDoubleSpinBox" name="refreshRateSpinBox">
              <property name="minimum">
               <double>0.500000000000000</double>
              </property>
              <property name="maximum">
               <double>5.000000000000000</double>
              </property>
              <property name="stepType">
               <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
              </property>
             </widget>
            </item>
            <item row="2" column="0" colspan="3">
             <widget class="QPushButton" name="toggleAutoRefreshButton">
              <property name="text">
               <string>自动刷新</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>刷新间隔</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="label_3">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>s</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QGridLayout" name="gridLayout_7" rowstretch="0,0,0,0,0" columnstretch="2,2,3">
            <item row="2" column="1">
             <widget class="QPushButton" name="pointARecordButton">
              <property name="text">
               <string>光标</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>A点</string>
              </property>
             </widget>
            </item>
            <item row="4" column="0" colspan="3">
             <widget class="QPushButton" name="calibrateButton">
              <property name="text">
               <string>标定波长</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>B点</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>波长 (nm)</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="2" column="2">
             <widget class="QDoubleSpinBox" name="pointAWaveLengthBox">
              <property name="maximum">
               <double>1000.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QPushButton" name="pointBRecordButton">
              <property name="text">
               <string>光标</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>位置</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item row="3" column="2">
             <widget class="QDoubleSpinBox" name="pointBWaveLengthBox">
              <property name="maximum">
               <double>1000.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="BasicSpectrumView" name="spectrumView"/>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>BasicSpectrumView</class>
   <extends>QGraphicsView</extends>
   <header>basicspectrumview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>refreshDevListButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>refreshDevList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>78</x>
     <y>246</y>
    </hint>
    <hint type="destinationlabel">
     <x>457</x>
     <y>342</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openDevButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>openCloseDevice(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>212</x>
     <y>246</y>
    </hint>
    <hint type="destinationlabel">
     <x>457</x>
     <y>342</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>refreshSpectrumButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateSpectrumData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>460</y>
    </hint>
    <hint type="destinationlabel">
     <x>457</x>
     <y>342</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toggleAutoRefreshButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleAutoUpdate(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>96</x>
     <y>515</y>
    </hint>
    <hint type="destinationlabel">
     <x>457</x>
     <y>342</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pointARecordButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>calibPointAChoose()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>79</x>
     <y>485</y>
    </hint>
    <hint type="destinationlabel">
     <x>544</x>
     <y>342</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pointBRecordButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>calibPointBChoose()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>79</x>
     <y>524</y>
    </hint>
    <hint type="destinationlabel">
     <x>544</x>
     <y>342</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>calibrateButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>doCalibrate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>110</x>
     <y>563</y>
    </hint>
    <hint type="destinationlabel">
     <x>544</x>
     <y>342</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>refreshDevList()</slot>
  <slot>openCloseDevice(bool)</slot>
  <slot>updateSpectrumData()</slot>
  <slot>toggleAutoUpdate(bool)</slot>
  <slot>calibPointAChoose()</slot>
  <slot>calibPointBChoose()</slot>
  <slot>doCalibrate()</slot>
 </slots>
</ui>
